Remove old receipt selector

tgolen opened the pull request:
### Details
The old receipt selector is no longer in use since the flows have been refactored. This PR removes the old files and incorporates the current receipt selector in the pages still referencing the old one.

### Fixed Issues
This is part of https://github.com/Expensify/App/issues/29107


### Tests
1. Create a `manual` money request (need to test this by requesting money from workspaces, existing users, and brand new users)
2. On the confirmation page, click the three dot menu and add a receipt
3. Verify it creates successfully
4. Create a `scan` money request
5. Verify it creates successfully
6. Create a `manual` money request without any receipt image
7. Click on the view for this request and click the image placeholder to add a receipt
8. Verify the receipt is properly added
9. Click on the receipt to go to the attachment view
10. Click on the three dot menu and select replace
11. Replace the receipt with another image and verify it is properly updated

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as the above

### QA Steps
Same as the above

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/2fc46138-f6a5-4db7-83c6-d735dae2ebeb)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
Unable to test due to broken emulator
</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/732d4965-12db-4e2d-96fb-f99fcc266a72)

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/a78269e2-6620-4142-8778-34299acf82bf)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/d9096727-ba5b-405f-aad9-50f49d0d8af8)

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
![image](https://github.com/Expensify/App/assets/1228807/8faa9a8d-982d-4ece-9449-d8997ab87146)

</details>

tgolen committed:
Remove old receipt selector files

tgolen committed:
Rename component references

tgolen committed:
Fix casing in import path

tgolen committed:
Use the new scan page instead of the old one

tgolen committed:
Use same redirect logic

tgolen committed:
Fix a mobile crash

johnmlee101 commented:
Testing, but also cc @ntdiary if you want to do the checklist for this PR since I see you're assigned on the parent issue

johnmlee101 commented:
For step 10 I get this menu option
<img width="1342" alt="image" src="https://github.com/Expensify/App/assets/5258036/d9aba815-c711-497e-89c7-47d950b2b8f7">

tgolen commented:
@johnmlee101 what's the URL of that page (and also the URL right before you get that)? I'll test that flow out again on my end.

johnmlee101 commented:
The IOU page
https://dev.new.expensify.com:8082/r/3641021911374988

Opening the image 
https://dev.new.expensify.com:8082/r/3641021911374988/attachment?source=https://www.expensify.com.dev/receipts/w_27ab9f28e2cc5041abf79d25a3c25aa6002e60e9.png

Clicking Replace Image
https://dev.new.expensify.com:8082/r/3641021911374988/edit/receipt

tgolen commented:
@situchan, it sounds like it would be helpful for you to take over the review of this. Thank you!

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
@johnmlee101 I was able to reproduce the bug you found. I'll have that fixed here in a few minutes I think

tgolen committed:
Fixed receipt replace option

tgolen committed:
Remove extra dependency

tgolen commented:
All right, I found the bug and fixed it so the replace flow should work again. Updated!

johnmlee101 commented:
retesting that flow real quick!

johnmlee101 reviewed (dismissed):
Nice, works 👍

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen committed:
Allow editing of receipt

tgolen commented:
I fixed conflicts and found a little bug with adding a receipt to an existing request that I fixed.

tgolen commented:
@situchan that change fixed a bug in the replace receipt flow. Because of the navigation happening there, `replaceReceipt()` is never called and the receipt isn't replaced.

I'm not sure what you are pointing out with the before/after. Is that the transaction object in Onyx? Is it a problem?

situchan commented:
Please check this video to see what I mean

https://github.com/Expensify/App/assets/108292595/3ae18adb-cf75-499a-a260-46e56976171c

situchan commented:
On main, `participants` data is consistent


https://github.com/Expensify/App/assets/108292595/af924148-042e-49aa-9357-d40bf456366f

tgolen commented:
OK, I think I'm following you now, thanks. I'll have a closer look at that and see what to do about it. It shouldn't be changing the participants.

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen committed:
Add an action param

tgolen committed:
Rename method and remove old method

tgolen committed:
DRY up code for mobile web

tgolen committed:
Rename method to be consistent

tgolen committed:
DRY up and make native file consistent

tgolen committed:
Use better logic for detecting global create

tgolen commented:
All right, I ended up making quite a few changes after digging into the `participants` problem.

1. I refactored the `index.native.js` file to be much more consistent with the `index.js` file
2. I refactored the URL routes to have either `/edit/...` or `/create/...` depending on what the user is doing
3. I started using `isFromGlobalCreate` instead of checking `report.reportID`
4. Using points 2 & 3, I changed the logic for what to do with a receipt and it is now much easier to follow

This will need full testing and review from the top again.

tgolen committed:
Fix comments and inverse logic

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
bump @johnmlee101 @situchan for review again. I fixed some conflicts.

johnmlee101 commented:
@situchan waiting for re-testing 👍

situchan commented:
retesting

tgolen commented:
How did your testing go? It would be great to get this merged soon.

situchan commented:
will complete checklist in an hr

situchan commented:
BUG1: 404 page when try to replace receipt


https://github.com/Expensify/App/assets/108292595/c6465973-167e-42e3-a143-d92edcf9f316

situchan commented:
I think the real bug is that manual request with receipt is not correctly created


https://github.com/Expensify/App/assets/108292595/965bb578-309a-43d6-9fa8-addc6f04fbc1

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen committed:
Add the backto parameter

tgolen commented:
OK, thanks. I'm going to dig into this and see what's going on.

On Tue, Dec 19, 2023 at 11:36 AM Situ Chandra Shil ***@***.***>
wrote:

> I think the real bug is that manual request with receipt is not correctly
> created
>
>
> https://github.com/Expensify/App/assets/108292595/965bb578-309a-43d6-9fa8-addc6f04fbc1
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Expensify/App/pull/32764#issuecomment-1863365376>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJMAB6BPMQPZDQSNEQB7DDYKHUEZAVCNFSM6AAAAABANJYQN6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRTGM3DKMZXGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

tgolen commented:
@situchan I've dug into that, and I've found a few things.

- **Primarily**: the error happening is reproducible in `main` with or without replacing a receipt (you can do it with just a normal manual request for $1). 
- However, I could never reproduce the error on staging :( 
- I am able to create a request with someone from the global create after I've send them at least one message
- The error is related to people in the "contacts" section of the participants list that have a null `reportID` property

# Conclusion
Since the same error happens on `main` and none of my PR changes how participants are added to the participant list, I am inclined to ask for a pass on this one. If we can narrow down the exact reproduction steps that lead to that network error, then I feel that a new GH should be opened to investigate it further.

For now, can you please test that flow with a user that is NOT in the "contact" portion of the participant list?

situchan commented:
If you see my video, I tested with existing (my own) workspace chat so requested money myself. It seems not relevant to participants list.
But agree out of scope as it already happens on main

tgolen commented:
I've reopened [some internal conversations](https://expensify.slack.com/archives/C049HHMV9SM/p1703102481084439?thread_ts=1688715731.926879&cid=C049HHMV9SM) that happened about 5 months ago related to that error. The discussion ended with no one being able to reliably reproduce it. However, I seem to be able to reproduce more than I am unable to reproduce it, so we'll see if we can find the source of the problem internally for now.

tgolen committed:
Lint

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
I've updated this with `main` and got the lint error fixed. Is there anything else before this can be merged?

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
Conflicts resolved and branch is updated to main. It would be great to have this merged! bump @johnmlee101 @situchan

situchan commented:
reviewing this too. Please complete missing author checklist items

tgolen commented:
Hm, I had them all checked off, but maybe the checklist was updated. I'll make sure I'm using the latest version.

situchan commented:
Found issue while executing

> 6. Create a manual money request without any receipt image
7. Click on the view for this request and click the image placeholder to add a receipt


This is similar [bug](https://github.com/Expensify/App/pull/32764#issuecomment-1863359083) I reported before but I am pretty sure this was caused by this PR as doesn't happen on main




https://github.com/Expensify/App/assets/108292595/a7e37e86-4da5-4828-951c-a8fb0a7b1718

tgolen committed:
Fix navigation for replacing receipt

tgolen commented:
Thank you for reporting that! It was indeed a bug in this PR and I found an old navigation path for that three-dot-menu. I have fixed it and updated this branch.

situchan commented:
yes, my next TODO after complete review of https://github.com/Expensify/App/pull/33303

situchan commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

situchan commented:
BUG: Not able to replace receipt on iOS


https://github.com/Expensify/App/assets/108292595/a58a52f6-da88-4802-9992-43f3f5f70e7f

situchan commented:
BUG: navigation is wrong when click back button on replace receipt page. Transaction detail page is opened twice


https://github.com/Expensify/App/assets/108292595/b2223e86-1486-4f8a-b3a0-c61918ef5011

production behavior:

https://github.com/Expensify/App/assets/108292595/a7c097d0-6ff4-4cb2-87cf-08faed673fd2

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen committed:
Fix navigation when pressing back from scan receipt modal

tgolen commented:
OK, I've fixed both of those. One of the flows needed an extra `dismisModal()`, one flow needed to have the `backTo` param removed from `goBack()` (because the app thought we were navigating forward for some reason). The receipts couldn't be replaced (or added) on iOS due to an error `[FileUtils] Could not read uploaded file [TypeError: Cannot read property 'replace' of undefined]`.

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

situchan commented:
main: Not found page issue doesn't happen for replace receipt page. But still happens on transaction detail page.
But both issue happen on this branch. I am not sure we should consider this out of scope.

<img width="334" alt="Screenshot 2024-01-04" src="https://github.com/Expensify/App/assets/108292595/504b5996-a062-4f4c-a49e-80ad4d3c5085">

https://github.com/Expensify/App/assets/108292595/4a34a020-7178-4cb7-82a5-e85683b5f522

tgolen commented:
OK, thank you. I will look into this further today.

tgolen commented:
Were there any other issues you encountered in your testing or is this the last bug?

situchan commented:
> Were there any other issues you encountered in your testing or is this the last bug?

I hope last bug 🙏 
Still keep testing various cases though

situchan commented:
Another bug is that I am not able to make successful request with camera scan at all. Tested on android

tgolen commented:
> Another bug is that I am not able to make successful request with camera scan at all. Tested on android

This is normal for the simulators though, right? I don't know of any way of testing the camera capture without a physical device. Also, isn't this also happening on `main`?

> It would have been ideal to fix not found page bug on transaction detail page before merging this PR so we can fully test without any known bugs.
> Do you know this GH? or not created yet?

Yeah, this is what I want to dig into. I don't really want to merge it with this known bug. The steps you took in the video to reproduce it are pretty clear, so that will help. I think that the problem lies in the optimistic data. From what I've seen, once the server response from the API comes back from creating the transaction, then everything is fine. The not-found page only exists during the period where the optimistic data is applied. I keep thinking there must be an incorrect reportID saved somewhere, but there are so many of them, it is very difficult to track down. I will continue to look into this and report back.

situchan commented:
> > Another bug is that I am not able to make successful request with camera scan at all. Tested on android
> 
> This is normal for the simulators though, right? I don't know of any way of testing the camera capture without a physical device. Also, isn't this also happening on `main`?

Ofc I tested on android physical device. Also happens on main.
So the bug here is that camera image always fails and backend sends error, while gallery image reaches "Receipt scan in progress" stage even for the same image.

tgolen commented:
Since I don't have an android device I can test this on, would you be able to debug what the issue could be? My guess is it's in here: https://github.com/Expensify/App/pull/32764/files#diff-21ba42adca2f3eb0a82245ae6bcc4614c06e9b4b7c208446ba9d6339d73e5b83R192 and maybe the source and filepath I changed are wrong?

tgolen commented:
FYI I was not able to dig into the not-found bug at all today (I had a meeting that went long). It'll be the first thing on my list tomorrow.

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
Diving into the not-found bug now.

tgolen commented:
Am I reproducing it correctly? For some reason, I'm not able to get the not-found page, and replacing the image works as expected.

https://github.com/Expensify/App/assets/1228807/a0a724fd-2870-4877-a105-f8c2d99feb80

situchan commented:
Please try to request money from IOU report > composer +, rather than from FAB.
Now I am not able to reproduce not found page anymore (instead, optimistic money request preview doesn't show) on main but still happens on this branch.

this branch:

https://github.com/Expensify/App/assets/108292595/bab04f43-bdf6-499e-a3a8-00fe54931fb2


main:

https://github.com/Expensify/App/assets/108292595/2e791d48-a9b3-4422-9476-c43e2614d718

situchan commented:
I reproduced not found page on transaction detail on android emulator (tested [this build](https://github.com/Expensify/App/pull/31558#issuecomment-1877471128))


https://github.com/Expensify/App/assets/108292595/5ecccc3e-17db-4a85-9f68-8561e0402b99

tgolen commented:
From looking at both of your videos for iOS and Android on my branch, it appears that the API command to create the transaction is failing (the report header being greyed out is the clue). In those cases, I think the not-found page is expected.

Can you confirm if you are indeed receiving API errors? I expect to find that you are getting the dreaded "there is already a report between these two people" error.

tgolen commented:
When I'm done with my meetings this morning I will try reproducing it from the in-composer + menu.

tgolen commented:
OK, I confirmed that I cannot reproduce it from the in-composer + menu either. I made sure I was fully up-to-date on our backend systems, so I propose that we merge this as-is. It seems like there is varying issues with some of the API requests and I'd like to officially get this on staging and production to see if we can figure out how to reliably reproduce it.

johnmlee101 commented:
I think I agree, let's get this out so we can focus on this with a bit more eyes and reproducibility

situchan commented:
ok, will checklist shortly

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
@situchan bump for checklist please

situchan commented:
Tests well. Please resolve conflict

tgolen committed:
Merge branch 'main' into tgolen-refactor-moneyrequestcreation-cleanup

tgolen commented:
Thanks! It's solved. Let's get this merged! @johnmlee101

situchan reviewed (approved):
I thoroughly tested but will still watch out any regressions we might have missed.

situchan commented:
TS failing

situchan commented:
There's TS check failure

tgolen committed:
Fix a couple bugs that TS found

tgolen commented:
Wow, nice that those were caught. I am guessing they cropped up from some other PR that got merged... Fixed!

tgolen commented:
quick! merge before something else conflicts!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7486823430) to staging by https://github.com/johnmlee101 in version: 1.4.24-4 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|cancelled 🔪
🕸 web 🕸|success ✅

Gonals commented:
@tgolen, I think this may have caused [this blocker](https://github.com/Expensify/App/issues/34417) (And likely [this](https://github.com/Expensify/App/issues/34388) one, although I can't reproduce it). I'm investigating what's going on

situchan commented:
> @tgolen, I think this may have caused [this blocker](https://github.com/Expensify/App/issues/34417) (And likely [this](https://github.com/Expensify/App/issues/34388) one, although I can't reproduce it). I'm investigating what's going on

yes, they're same bug and main test flow in this PR and we demoted as happening on production (not reliably reproducible)
https://github.com/Expensify/App/pull/32764#issuecomment-1863359083
https://github.com/Expensify/App/pull/32764#issuecomment-1863365376

Gonals commented:
> > @tgolen, I think this may have caused [this blocker](https://github.com/Expensify/App/issues/34417) (And likely [this](https://github.com/Expensify/App/issues/34388) one, although I can't reproduce it). I'm investigating what's going on
> 
> yes, they're same bug and main test flow in this PR and we demoted as happening on production (not reliably reproducible) [#32764 (comment)](https://github.com/Expensify/App/pull/32764#issuecomment-1863359083) [#32764 (comment)](https://github.com/Expensify/App/pull/32764#issuecomment-1863365376)

Oh, gotcha. I haven't been able to reproduce in prod, but it consistently happens in main/staging. Good to know it is not a blocker, tho

joelbettner commented:
@tgolen I think the changes here might be related to https://github.com/Expensify/App/issues/34295

situchan commented:
> @tgolen I think the changes here might be related to #34295

Hmm, I don't think so. I tested with receipt image and worked fine. This PR didn't touch such logic of pdf not working while image working. Jack requested that pdf to reproduce but no answer yet.

sample:
![receipt_sample](https://github.com/Expensify/App/assets/108292595/57c8cf84-2c29-48fd-b538-d8866ce888d8)

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7528224417) to production by https://github.com/mountiny in version: 1.4.24-8 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

DylanDylann commented:
@tgolen In this PR, we use `action` param in `IOURequestStepScan` but in the creation flow the `action` param always be `undefined`. See below video

https://github.com/Expensify/App/assets/141406735/31b3892d-8158-4007-bcf1-93eb63c167b0

It was caused by we missed updating the route to use `action` param here https://github.com/Expensify/App/blob/ef151c3871724bf9de882cb77cd7e71682a27121/src/ROUTES.ts#L396-L400 
And we also need to update new route for `IOURequestStartPage`

cc @situchan

truph01 commented on src/pages/iou/request/step/IOURequestStepScan/index.native.js:
```diff
@@ -118,7 +124,57 @@ function IOURequestStepScan({
 
-    const takePhoto = useCallback(() => {
+    const navigateBack = () => {
+        Navigation.goBack();
```
This PR dropped the backTo parameter on native — which causes the bug https://github.com/Expensify/App/issues/82491
